hydra-node/src/Hydra/Chain/Direct/Context.hs

```diff
@@ -197,14 +197,16 @@ genStClosed ctx utxo = do
   (u0, stOpen) <- genStOpen ctx
   confirmed <- arbitrary
   let (sn, snapshot, toFanout) = case confirmed of
-        cf@InitialSnapshot{snapshot = s} ->
+        InitialSnapshot{} ->
           ( 0
-          , cf{snapshot = s{utxo = u0}}
+          , InitialSnapshot {initialUtxo = u0}
           , u0
           )
-        cf@ConfirmedSnapshot{snapshot = s} ->
-          ( number s
-          , cf{snapshot = s{utxo = utxo}}
+        ConfirmedSnapshot{snapshot = snap, signatures} ->
+          ( number snap
+          , ConfirmedSnapshot
+            { snapshot = snap { utxo = utxo }
+            , signatures}
           , utxo
```

Comment:
I agree. This PR paves the way for refactoring that arguably ugly part

Reply:
I think there's a need to change the call sites, perhaps unbundling the UTxO and signatures when constructing the Tx

Reply:
I tried to get a clue of what this conversation is about to improve the PR but I must admit that, at this moment, I missing to much context to have any useful contribution here.

So I'll mark the PR as ready and I let you guys decide if we should merge as is or spend some time pairing to improve it.

Reply:
That's a good strategy. The problem we are alluding at here is that the 2 different cases of "confirmed" snapshots need to be handled in our tests as they represent 2 different paths towards fanout and it's important to cover because `U0` is a special case. But the way we handled it was clumsy as you've seen, leading to "Impossible states being representable". 
Having sorted out this problem whereby an initial snapshot does not have a number nor a bunch of signatures, we should be in a better position to simplify this code and make it more explicit.